lib/class-wp-block-type.php

```diff
@@ -149,6 +149,16 @@ public function prepare_attributes_for_render( $attributes ) {
 			$prepared_attributes[ $attribute_name ] = $value;
 		}
 
+		/**
+		* Filter to allow plugins to override Block attributes.
+		* 
+		* @since 3.5.0
+		* 
+		* @param array $prepared_attributes Block attributes.
+		* @param WP_Block_Type $this Block object which attributes are filtered.
+		*/
+		$prepared_attributes = apply_filters( 'block_attributes', $prepared_attributes, $this );
```

Comment:
It seems like this name should be more specific, `block_render_attributes` or `block_prepared_attributes`? Otherwise, it won't be clear at which stage those attributes are going to be modified. You need also to take into account that in the future, we might want to allow to filter block attributes registered on the server before we send them to the client.

Reply:
OK, I agree, renamed to block_prepared_attributes.

about #8912 I have pinged Bruce who pushed that PR to be aware I am proposing a filter which makes his proposal a bit redundant. (this comment is just a FYI)